modules/copy-post.php

```diff
@@ -0,0 +1,190 @@
+            'post_excerpt' => $source_post->post_excerpt,
+            'post_category' => $source_post->post_category,
+            'tags_input' => $source_post->tags_input,
+        ) );
+        return wp_update_post( $data );
+    }
+
+    /**
+     * Update the target post's featured image.
+     *
+	 * @param WP_Post   $source_post Post object to be copied.
+     * @param int       $target_post_id Target post ID.
+     * @return int|bool Meta ID if the key didn't exist, true on successful update, false on failure.
+     */
+    protected function update_featured_image( $source_post, $target_post_id ) {
+        $featured_image_id = get_post_thumbnail_id( $source_post );
+        return update_post_meta( $target_post_id, '_thumbnail_id', $featured_image_id );
+    }
+
+    /**
+     * Update the target post's post format.
+     *
+     * @param WP_Post               $source_post Post object to be copied.
+     * @param int                   $target_post_id Target post ID.
+     * @return array|WP_Error|false WP_Error on error, array of affected term IDs on success.
+     */
+    protected function update_post_format( $source_post, $target_post_id ) {
+        $post_format = get_post_format( $source_post );
+        return set_post_format( $target_post_id, $post_format );
+    }
+
+    /**
+     * Update the target post's title.
+     *
+     * @param string  $post_title Post title determined by `get_default_post_to_edit()`.
+     * @param WP_Post $post       Post object of newly-inserted post.
+     * @return string             Updated post title from source post.
+     */
+    function filter_title( $post_title, $post ) {
+        return $post->post_title;
+    }
+
+    /**
+     * Update the target post's content (`post_content`).
+     *
+     * @param string  $post_content Post content determined by `get_default_post_to_edit()`.
+     * @param WP_Post $post         Post object of newly-inserted post.
+     * @return string               Updated post content from source post.
+     */
+    function filter_content( $post_content, $post ) {
+        return $post->post_content;
+    }
+
+    /**
+     * Update the target post's excerpt.
+     *
+     * @param string  $post_excerpt Post excerpt determined by `get_default_post_to_edit()`.
+     * @param WP_Post $post         Post object of newly-inserted post.
+     * @return string               Updated post excerpt from source post.
+     */
+    function filter_excerpt( $post_excerpt, $post ) {
+        return $post->post_excerpt;
+    }
+
+    /**
+     * Add a "Copy" row action to posts/pages/CPTs on list views.
+     *
+     * @param array   $actions Existing actions.
+     * @param WP_Post $post    Post object of current post in list.
+     * @return array           Array of updated row actions.
+     */
+    function add_row_action( $actions, $post ) {
+        $edit_url = add_query_arg( array(
+            'post_type' => $post->post_type,
```

Comment:
I wonder if we should limit this to `post` for now. CPTs often have a more complicated structure, and tend to make use of custom taxonomies, post meta, and other things we may not support yet.

This would have the added benefit of not conflicting with another plugin's duplication feature, WooCommerce:
![image](https://user-images.githubusercontent.com/426388/51336004-cd5d6a00-1a83-11e9-83f9-1a0a213ac4cf.png)

Reply:
Maybe posts and pages?

Reply:
Hm, let me see if I can move to just supporting a fixed list of post types (posts, pages, JP CPTs?), seeing if the elements are all there for them.

Reply:
That sounds good. You could make that filterable as well, so folks could add support later on for their own CPTs.